src/trans.cc

```diff
@@ -4623,169 +4623,39 @@ Obj PowTrans4Perm4(Obj f, Obj p)
 ** Quotient a transformation f by a permutation p: f * p ^ -1
 *******************************************************************************/
 
-Obj QuoTrans2Perm2(Obj f, Obj p)
+static Obj QuoTrans2Perm2(Obj f, Obj p)
 {
-    UInt    def, dep, i;
-    const UInt2 * ptf, *ptp;
-    UInt2 * ptquo;
-    UInt4 * pttmp;
-    Obj     quo;
-
-    def = DEG_TRANS2(f);
-    dep = DEG_PERM2(p);
-    quo = NEW_TRANS2(MAX(def, dep));
-    ResizeTmpTrans(SIZE_OBJ(p));
-
-    // invert the permutation into the buffer bag
-    pttmp = ADDR_TRANS4(TmpTrans);
-    ptp = CONST_ADDR_PERM2(p);
-    for (i = 0; i < dep; i++) {
-        pttmp[*ptp++] = i;
-    }
-
-    ptf = CONST_ADDR_TRANS2(f);
-    ptquo = ADDR_TRANS2(quo);
-
-    if (def <= dep) {
-        for (i = 0; i < def; i++) {
-            *(ptquo++) = pttmp[*(ptf++)];
-        }
-        for (i = def; i < dep; i++) {
-            *(ptquo++) = pttmp[i];
-        }
-    }
-    else {
-        for (i = 0; i < def; i++) {
-            *(ptquo++) = IMAGE(ptf[i], pttmp, dep);
-        }
-    }
-    return quo;
+    Obj inv = INV(p);
+    GAP_ASSERT(TNUM_OBJ(inv) == T_PERM2);
+    return ProdTrans2Perm2(f, inv);
 }
 
-Obj QuoTrans2Perm4(Obj f, Obj p)
+static Obj QuoTrans2Perm4(Obj f, Obj p)
 {
```

Comment:
My longterm plan is to make all functions `static` that are not explicitly for external use, so as to avoid the situation were people "accidentally" use them w/o being aware of them being private. See also PR #3204. That doesn't mean they have to stay static/private, but if somebody needs to use a function, we (well, me, at least) would prefer to know about it explicitly. That frees us from the burden of having to verify for every single kernel refactoring whether maybe some package is using that interface anyway (though, luckily, we now again compile and at least test-load almost every distributed package again, including digraphs and semigroups, so this is less of a problem than it used to be).

The other advantage is that it may help the compiler with some optimizations and allows for a slightly slimmer binary (but these are relatively minor points).

BTW, if you have any need to invoke any of these function directly from, say, semigroups (instead of going via e.g. `QUO` in this case), please tell us, then of course we can leave of the `static` from those, and/or find some other means to accommodate you; the goal here really isn't to annoy anybody, but rather to prevent future annoyances.